### PR TITLE
changes for website

### PR DESCRIPTION
**Pull request by duncathan**
- command to extract readme sections
- command to extract db videos in yaml format
- command to extract game data
- games have a new field for website info

### CODE REVIEW

**Comment by duncathan** [comment]
let's not worry about the fact that i somehow based this branch on a commit from exactly three months ago

File: pyproject.toml [hunk 1]
```diff
@@ -65,8 +65,9 @@ dependencies = [
     # error reports
     "sentry-sdk",
 
-    # Database Video html Site Generator 
+    # Database Video html Site Generator
     "htmlmin",
+    "ruamel.yaml",
```
Comment (henriquegemignani) [inline pyproject.toml:70]: I wonder if these two could be an extra which we don't include in the executable.

File: randovania/cli/commands/export_db_videos.py [hunk 2]
```diff
@@ -283,6 +286,59 @@ def filename_friendly_game_name(game: RandovaniaGame):
...
+                if sorted_node:
+                    add_entry(sorted_area, node, sorted_node)
+
+    output = {"regions": output}
+
+    tr = lambda s: s
+    fmt = "yml"
+    if as_frontmatter:
+        tr = lambda s: f"---\n{s}\n---"
+        fmt = "yml"
```
Comment (henriquegemignani) [inline randovania/cli/commands/export_db_videos.py:335]: Is this supposed to be the same extension?

File: randovania/games/prime2/game_data.py [hunk 3]
```diff
@@ -111,6 +111,19 @@ def _exporter():
...
+    web_info=game.GameWebInfo(
+        what_can_randomize=[
+            "All items including Temple Keys",
+            "Elevator destinations",
+            "Starting locations",
+            "Door locks",
+            "Translator gate requirements",
+        ],
+        need_to_play=[
+            "An ISO of any Gamecube release of the game",
```
Comment (henriquegemignani) [inline randovania/games/prime2/game_data.py:123]: NTSC and PAL regions only. No Dark Echoes allowed.

File: randovania/games/game.py [hunk 4]
```diff
@@ -96,6 +96,21 @@ class GameGenerator:
     """Use AllJokesDistributor if not using hints."""
 
 
+@dataclass(frozen=True)
+class GameWebInfo:
+    what_can_randomize: Iterable[str] = (
+        "Everything",
```
Comment (henriquegemignani) [inline randovania/games/game.py:102]: These values fit best as values for Blank Game, instead of being default values.

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

File: randovania/games/prime2/game_data.py (hunk 3 above)
Comment (Miepee) [reply randovania/games/prime2/game_data.py:123]: Huh, i thought JP was also allowed.

File: randovania/cli/commands/export_db_videos.py [hunk 5]
```diff
@@ -283,6 +286,59 @@ def filename_friendly_game_name(game: RandovaniaGame):
     return "".join(x for x in game.long_name if x.isalnum() or x in [" "])
 
 
+def export_as_yaml(game: RandovaniaGame, out_dir: Path, as_frontmatter: bool):
```
Comment (Miepee) [inline randovania/cli/commands/export_db_videos.py:289]: Not gonna point out every single one of them: are all the mypy errors intended?

**Review by Miepee — COMMENTED** [review]
Our CLI commands usually have unit tests.

File: randovania/cli/commands/export_db_videos.py (hunk 2 above)
Comment (duncathan) [reply randovania/cli/commands/export_db_videos.py:335]: no oops

File: randovania/cli/commands/export_db_videos.py (hunk 5 above)
Comment (duncathan) [reply randovania/cli/commands/export_db_videos.py:289]: yes, insofar as virtually every other CLI function would have the same errors and it seems out of scope to fix them all

File: pyproject.toml (hunk 1 above)
Comment (duncathan) [reply pyproject.toml:70]: certainly could be, although out of scope for this pr imo

Comment (henriquegemignani) [reply pyproject.toml:70]: I'd prefer that done before the next release though.

Comment (duncathan) [reply pyproject.toml:70]: kk. you'll have to show mii how to do it

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**